apps/studio/src/features/editing-experience/components/MoveResourceModal/MoveResourceModal.tsx

```diff
@@ -191,21 +200,42 @@ const MoveResourceContent = withSuspense(
                 </Flex>
               )}
               {data?.pages.map(({ items }) =>
-                items.map((child) => {
+                items.map((item) => {
+                  const isItemDisabled: boolean =
+                    item.id === movedItem?.resourceId
+                  const isItemHighlighted: boolean =
+                    isResourceHighlighted && item.id === curResourceId
+
                   return (
                     <MoveItem
-                      {...child}
-                      isDisabled={child.id === movedItem?.resourceId}
-                      key={child.id}
-                      onChangeResourceId={() => {
-                        setResourceStack((prev) => [
-                          ...prev,
-                          {
-                            ...child,
-                            parentId: curResourceId ?? null,
-                            resourceId: child.id,
-                          },
-                        ])
+                      {...item}
+                      key={item.id}
+                      isDisabled={isItemDisabled}
+                      isHighlighted={isItemHighlighted}
+                      handleOnClick={() => {
+                        if (isItemDisabled) {
+                          return
+                        }
```

Comment:
nit: i think we can omit this case cos if it's disabled the `onClick` for the button shouldn't fire

Reply:
yeah true, i added it because it will be less confusing to future code maintainer by explicitly accounting for all states